inject/src/main/java/io/micronaut/inject/beans/visitor/IntrospectedTypeElementVisitor.java

```diff
@@ -180,6 +180,8 @@ private void processIntrospected(ClassElement element, VisitorContext context, A
         Introspected.AccessKind[] accessKinds = introspected.enumValues("accessKind", Introspected.AccessKind.class);
         Introspected.Visibility[] visibilities =
                 introspected.enumValues("visibility", Introspected.Visibility.class);
+        final String introspectionPackage = introspected.stringValue("introspectionPackage").orElse(element.getPackageName());
```

Comment:
If this package names differ we need to change the default visibility to only public members (see `Visibility.PUBLIC`) since package private and protected members will no longer be visible from different packages. In the 4.0.x branch we support reflection as a fall back but not here.